### ISSUE TITLE
REGRESSION: Can no longer get return value from C# method when calling from GDScript.

### ISSUE THREAD

**Issue by Ryan-000** [godotengine/godot#89584]
### Tested versions

4.2.2.rc2.mono (Not present in 4.2.2.rc1.mono)

### System information

Desktop

### Issue description

I can't get the return value of C# methods anymore when I updated to 4.2.2.rc2. Doesn't matter if it's a preloaded script, or used with [GlobalClass]

(Using. the .call("Add", 1, 2) does work but its not ideal).
![image](https://github.com/godotengine/godot/assets/73148864/d0cf48f3-df4a-4166-a673-120303b2b78a)

It works in 4.2.2.rc1
![image](https://github.com/godotengine/godot/assets/73148864/bb3f5192-4851-423e-823e-0450f6b094c9)

```gdscript
extends Node

const SomeCsharpScript = preload("res://SomeCsharpScript.cs")
var instance: SomeCsharpScript = SomeCsharpScript.new()

func _ready() -> void:
	var result = instance.Add(1, 2) # does not work in 4.2.2.rc2.mono
	print(result)

```

### Steps to reproduce

use mrp

### Minimal reproduction project (MRP)

[MRP.zip](https://github.com/godotengine/godot/files/14625205/MRP.zip)

**Comment by raulsntos** [comment]
- Should be fixed by https://github.com/godotengine/godot/pull/86972, it hasn't been cherry-picked yet.

**Comment by akien-mga** [comment]
Fixed by #86972, I cherry-picked it for 4.2.2-stable.
